src/pages/nutrition/table.jsx

```diff
@@ -0,0 +1,149 @@
+    },
+    {
+      off_nutriment_id: "energy_kcal",
+      label: "Fat",
+      value: "",
+      unit: null,
+      quantification: "<",
+      robotoffPrediction: null
+    }
+  ])
+
+  const [additionalNutriments, setAdditionalNutriments] = React.useState([
+    {
+      off_nutriment_id: "energy_kj",
+      label: "Energie (kJ)",
+      value: "",
+      unit: "null",
+      quantification: "=",
+      robotoffPrediction: null
+    },
+    {
+      off_nutriment_id: "energy_kcal",
+      label: "Protein",
+      value: "",
+      unit: "null",
+      quantification: "<",
+      robotoffPrediction: null
+    }
+  ])
+
+  function onchangeHandler(e) {
+    const {value, name} = e.target
+    setNutriments(prevState => prevState.map(
+      nutr => {
+        return name === nutr.label? {...nutr, value} : nutr
+      }
+    ))
+  }
+
+  function deleteItem(nutrition) {
+    setNutriments(prev => prev.filter(elem => elem !== nutrition))
+    setAdditionalNutriments(prev => [...prev, nutrition])
+  }
+
+  const rows = nutriments.map(nutrition => {
+    return (
+      createData(<Box sx={{
+          display: "flex",
+          alignItems: "center",
+          flexDirection: "row"
+        }}><TextField id="outlined-basic"
+                      type={'number'}
+                      label={nutrition.label}
+                      variant="outlined"
+                      sx={{ width: "10rem" }}
+                      value={nutrition.value}
+                      name={nutrition.label}
+                      onChange={onchangeHandler}
+        /> , <SelectAutoWidth /></Box>,
+        <DeleteOutlineIcon sx={{cursor: 'pointer', color: 'red'}} onClick={() => deleteItem(nutrition)}/>));
+  });
+
+  return (
+    <Box>
+      <TableContainer sx={{ margin: 0, maxWidth: "1000px", width: "340px" }}>
+        <Table aria-label="simple table">
+          <TableHead>
+            <TableRow>
+              <TableCell sx={{
+                maxWidth: "8em",
+                fontSize: "large",
+                fontWeight: "bold"
+              }}>nutrition.table.value</TableCell>
+
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {rows.map((row) => (
+              <TableRow
+                key={row.label}
+                sx={{ "&:last-child td, &:last-child th": { border: 0 } }}
+              >
+                <TableCell component="th"
+                           scope="row"
+                           sx={{ width: 10 }}
+                >
+                  {row.label}
+                </TableCell>
+                <TableCell align="left"
+                           sx={{ width: "1rem"}}>{row.property}</TableCell>
+              </TableRow>
+            ))}
+          </TableBody>
+        </Table>
+      </TableContainer>
+      <AdditionalNutriments
+        options = {additionalNutriments}
+        setNutriments={setNutriments}
+        setAdditionalNutriments={setAdditionalNutriments}
```

Comment:
You are trying to keep up to date 2 different arrays which is error pron. As soon as you forget an edge case, the two arrays will not be sync and it's hard to debug

I propose to keep the `setNutriments` that could be renamed `addNutrimentToTheTable`
Replace the `setAdditionalNutriments` by `missingNutriments` which would be computed as follow:

```js
const missingNutriments = offNutriments.filter(nutrimentKey => 
	data.every(element) => element.off_nutriment_id !== nutrimentKey) 
)
```

with `offNutriments` the list of all the nutrients available on openfoodfact

This would replace `options` and `setAdditionalNutriments`